test/plugin/test_out_google_cloud_grpc.rb

```diff
@@ -224,12 +224,20 @@ def api_client
 
   # GRPC logging mock that successfully logs the records.
   class GRPCLoggingMockService < Google::Logging::V1::LoggingService::Service
-    def initialize(requests_received)
+    def initialize(stub_params, requests_received)
       super()
       @requests_received = requests_received
+      @expected_log_names = stub_params.map do |param|
+        "projects/#{param[:project_id]}/logs/#{param[:log_name]}"
+      end
     end
 
     def write_log_entries(request, _call)
+      matched = @expected_log_names.select do |expected|
+        expected == request.log_name
+      end
+      fail GRPC::BadStatus.new(99, "Unexpected request: #{request.inspect}") \
+        if matched.empty?
```

Comment:
In fact, now that we're doing straight-up string equality, this is just:
```
fail GRPC::BadStatus.new(99, "Unexpected request: #{request.inspect}") \
  unless @expected_log_names.include?(request.log_name)
```
which lets you get rid of the whole `matched` bit.